openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -920,6 +924,23 @@ Below is a non-normative example of a `proof` parameter:
 
 ```
 
+#### `attestation` Proof Type {#attestation-proof-type}
+
+A key attestation in JWT format as defined in (#keyattestation-jwt).
+
+When a key attestation is used as a proof type, it MUST contain the `c_nonce` value provided by the Credential in its `nonce` parameter.
```

Comment:
A key attestation will be generated by the wallet provider and typically not by the wallet instance, right? If so, this requires interaction with the wallet provider when requesting credentials, without the possibility to obtain the attestation before hand.
This seems to introduce extra load and availability requirements on the wallet provider, as well as a slight loss of privacy by the wallet instance

Reply:
The tradeoff between `attestation` and `jwt` proof type:
- `attestation` must be done ad-hoc because the nonce is mandatory (if issues by wallet provider this includes a round-trip to the backend service) but does not require a proof of possession and therefore no user interaction
- `jwt` may include an key attestation (that may already be fetched) but requires proof of possession and therefore potentially user interaction (which may be bad for refresh/batch issuance)

Reply:
@pmhsfelix for pre-obtained key attestations, you would use jwt proof type and put key attestation in the header. so that the freshness is guaranteed by nonce in the jwt proof type but key attestation can be generated beforehand.

@paulbastian bringing up description of what is key attestation and how there are two types of key attestation before jwt and key attestation proof types are described and then describe jwt/attestation proof types might help prevent this kind of confusion - I had exctly the same confusion as Pedro, initially.

Reply:
I had the same confusion and I agree that explaining it earlier would really help! And adding @paulbastian's comment as spec text would help as well.

Reply:
I may be still missing something. Just to confirm: 
`attestation` proof-type 
- requires `nonce`, therefore needs to be built just-in-time for the credential request?
- requires interaction with the wallet provider to be build?
If so, this requires interaction with the wallet provider just immediately before credential request, right?
If not, what am I missing?